ext/standard/dir.c

```diff
@@ -189,21 +156,71 @@ PHP_FUNCTION(dir)
 }
 /* }}} */
 
+
+static php_stream* php_dir_get_directory_stream_from_user_arg(zval *arg)
+{
+	zend_resource *res;
+	if (arg == NULL) {
+		if (UNEXPECTED(DIRG(default_dir) == NULL)) {
+			zend_type_error("No resource supplied");
+			return NULL;
+		}
+		res = DIRG(default_dir);
+	} else {
+		ZEND_ASSERT(Z_TYPE_P(arg) == IS_RESOURCE);
+		res = Z_RES_P(arg);
+	}
+
+	if (UNEXPECTED(res->type != php_file_le_stream())) {
+		zend_argument_type_error(1, "must be a valid Directory resource");
+		return NULL;
+	}
+	php_stream *dir_stream = (php_stream*) res->ptr;
+	if (UNEXPECTED((dir_stream->flags & PHP_STREAM_FLAG_IS_DIR)) == 0) {
+		zend_argument_type_error(1, "must be a valid Directory resource");
+		return NULL;
+	}
+	return dir_stream;
+}
+
+static php_stream* php_dir_get_directory_stream_from_this(zval *this_z)
+{
+	zval *handle_zv = Z_DIRECTORY_HANDLE_P(this_z);
+	if (UNEXPECTED(Z_TYPE_P(handle_zv) != IS_RESOURCE)) {
+		zend_throw_error(NULL, "internal directory stream has been altered");
```

Comment:
Shouldn't this be a type error (same below)? Doesn't matter much.
Also shouldn't the error messages start with capital letter (as they're not argument errors)

Reply:
I don't know if this should be a TypeError, as normally a non-initialized object is an `Error`

Reply:
It doesn't matter much, fine by me.